tests/grid/tria_clear_user_data.cc

```diff
@@ -0,0 +1,46 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 1998 - 2018 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE.md at
+// the top level directory of deal.II.
+//
+// ---------------------------------------------------------------------
+
+
+// We assess the possibility of using clear user data in any dim.
+
+#include <deal.II/grid/grid_generator.h>
+#include <deal.II/grid/grid_out.h>
+#include <deal.II/grid/tria.h>
+
+#include "../tests.h"
+
+template <int dim>
+void
+test()
+{
+  Triangulation<dim> tria;
+  GridGenerator::hyper_cube(tria);
+  deallog << 'Cleared user data in ' << dim << 'D' << std::endl;
```

Comment:
Don't you actually want to call the function you are testing?

Reply:
whops, forgot one line. thanks

Reply:
To fix the error on the test system,
```
/jenkins/workspace/dealii_PR-12302/tests/grid/tria_clear_user_data.cc:32:14: error: character constant too long for its type [-Werror]
   deallog << 'Cleared user data in ' << dim << 'D' << std::endl;
```
please change this into
```suggestion
  deallog << "Cleared user data in " << dim << 'D' << std::endl;
```